📝 Add docs for OTel header generation and ingestion (#302)
DataDog/dd-sdk-flutter: packages/datadog_flutter_plugin/doc/rum/otel_support.md

# RUM Flutter Open Telemetry Support

> The [Datadog Tracking Http Client][1] package and [gRPC Interceptor][2] package both support distributed traces through both automatic header generation and header ingestion.

Reviewer: Header ingestion?

Author: Good catch, I wrote this before I removed the ability for the client to parse and forward OTel headers.

## Datadog Header Generation

When configuring your tracking client or gRPC Interceptor, you can specify the types of tracing headers you want Datadog to generate. For example, if you want to send `b3` headers to `example.com` and `tracecontext` headers for `myapi.names`, you can do so with the following code:

```dart
final hostHeaders = {
    'example.com': { TracingHeaderType.b3 },
    'myapi.names': { TracingHeaderType.tracecontext}
};
```

You can use this object during initial configuration:

```dart
// For default Datadog HTTP tracing:
final configuration = DdSdkConfiguration(
    // configuration
    firstPartyHostsWithTracingHeaders: hostHeaders,
);
```

Then enable tracing as normal.

> Note that this information will be merged with any hosts set on `DdSdkConfiguration.firstPartyHosts`. Hosts specified in `firstPartyHosts` will generate Datadog Tracing Headers by default.

Reviewer 2: ```suggestion
This information is merged with any hosts set on `DdSdkConfiguration.firstPartyHosts`. Hosts specified in `firstPartyHosts` generate Datadog Tracing Headers by default.
```
